### DISCUSSION TITLE
ExportAction Duplicate record row

### DISCUSSION THREAD

**Discussion by waltermiglioress** [filamentphp/filament#12252] [Help]
### Package

Actions

### Package Version

v3.2.3

### How can we help you?

Hello everyone,
is there a way to duplicate a calculated record inside the Exporter?
I have 3 cols (start_date,end_date, value). I need to split the value between the range in months. I already calculated the difference in months, but now I don't know how to split the value creating n rows from value / (int) period.

```php
ExportColumn::make('months')
                ->state(function (Production $record): string {
                    $date_start = !is_null($record->date_start ?? null) ? Carbon::parse($record->date_start) : null;
                    $date_end = !is_null($record->date_end ?? null) ? Carbon::parse($record->date_end) : null;
                    $diffInMonths = $date_start ? $date_start->diffInMonths($date_end) : 1;
                    $subd = ((int)$diffInMonths > 1) ? $record->value/ (int)$diffInMonths : $record->value;

                    return $subd;
                    })
``` 
Like this, I calculate and added only the additional column while what I need is that under the main row with the value appears as many lines as are those calculated based on the difference in months, without saving them on the db, just for the exportation only in order to provide a different rappresentation (in vertical view and not in columns wich provide an horizontal view of the data).
Sorry for my english mistakes. I hope i was clear.

Thanks at all for the support in advance.

**Comment by m4teches** [comment]
Hi, did you manage to achieve this?

**Reply by waltermiglioress** [reply]
Yes!I had to change the approach. I made a controller with a function export for managing the download through the Excel package  from Maatwebsite.


    $ids = $request->query('ids');
    // Log::info("IDs received for export: " . $request->query('ids'));

    if ($ids) {
        $idsArray = explode(',', $ids); // Convert the string of IDs into an array
        $query = \App\Models\Production::whereIn('id', $idsArray);
    } else {
        $query = \App\Models\Production::query();  // If there are no IDs, export everything
    }

    $export = new ProductionExport($query);
    $fileName = 'productions-' . now()->format('Y-m-d_His') . '.xlsx';
    return Excel::download($export, $fileName);


And a ProductionExport that manage the file with its styles, column formats, headings and of course the core for the calculated rows in a map function.

```
public function map($row): array
{
    $rows = [];
    $start = Carbon::parse($row->date_start);
    $end = Carbon::parse($row->date_end);

    // Create an array to store all months covered by the interval
    $currentMonth = $start->copy()->startOfMonth();
    $endMonth = $end->copy()->startOfMonth();
    $months = [];

    while ($currentMonth->lessThanOrEqualTo($endMonth)) {
        $months[] = $currentMonth->copy(); // Add the current month to the array
        $currentMonth->addMonth(); // Move to the next month
    }

    Log::info('After the while loop, months: ' . $currentMonth);

    foreach ($months as $month) {
        $monthlyTaxableAmount= $row->taxable_amount / count($months);

        $rows[] = [
            $row->desc,
            $row->type,
           .....
            $row->taxable_amount ,
            $start->format('d/m/Y'),
            $end->format('d/m/Y'),
//          $row->date_start->format('d/m/Y'),
//          $row->date_end->format('d/m/Y'),
            $row->status,
            $row->client->name,
            $month->format('d/m/Y'),
            $monthlyTaxableAmount
        ];
    }
    return $rows;
}
```

I called the Controller on the action 
```
->action(function (Collection $records) {
    $ids = $records->pluck('id')->join(',');
    Log::info("IDs received from bulk: " . $ids);
    
    $c = new ProductionController(); // Extract the IDs and join them into a comma-separated string
    return $c->export(new Request(['ids' => $ids]));
})
```
I hope this can help you.

**Reply by m4teches** [reply]
Nice workaround, so you used a bulk action. I wanted to achieve this by the default export action but looks like I have to use a different method.
